### PR TITLE
LWDEV-9158 New logging for job (do not merge without 9159).

### PR DESCRIPTION
**Pull request by dmitrysmall**
Please, see: https://lykkex.atlassian.net/browse/LWDEV-9158.

### CODE REVIEW

File: src/Lykke.Job.Stellar.Api/Startup.cs [hunk 1]
```diff
@@ -62,21 +61,47 @@ public IServiceProvider ConfigureServices(IServiceCollection services)
...
 
                 var builder = new ContainerBuilder();
-                var appSettings = Configuration.LoadSettings<AppSettings>();
-
-                Log = CreateLogWithSlack(services, appSettings);
+                var appSettings = Configuration.LoadSettings<AppSettings>(options =>
+                {
+                    options.SetConnString(x => x.SlackNotifications.AzureQueue.ConnectionString);
+                    options.SetQueueName(x => x.SlackNotifications.AzureQueue.QueueName);
+                    options.SenderName = "JobStellarApi";
```
Comment (KonstantinRyazantsev) [inline src/Lykke.Job.Stellar.Api/Startup.cs]: Use AppEnvironment.Name

File: src/Lykke.Job.Stellar.Api/Startup.cs [hunk 2]
```diff
@@ -105,139 +130,66 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env, IApplica
...
             catch (Exception ex)
             {
-                Log?.WriteFatalErrorAsync(nameof(Startup), nameof(Configure), "", ex).GetAwaiter().GetResult();
+                Log?.Critical(ex);
                 throw;
             }
         }
 
-        private async Task StartApplication()
+        private void StartApplication()
```
Comment (KonstantinRyazantsev) [inline src/Lykke.Job.Stellar.Api/Startup.cs]: Why this method made sync?

File: src/Lykke.Job.Stellar.Api/Startup.cs [hunk 3]
```diff
@@ -105,139 +130,66 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env, IApplica
...
             catch (Exception ex)
             {
-                await Log.WriteFatalErrorAsync(nameof(Startup), nameof(StartApplication), "", ex);
+                Log.Critical(ex);
                 throw;
             }
         }
 
-        private async Task StopApplication()
+        private void StopApplication()
```
Comment (KonstantinRyazantsev) [inline src/Lykke.Job.Stellar.Api/Startup.cs]: And this

File: src/Lykke.Job.Stellar.Api/Startup.cs (hunk 2 above)
Comment (dmitrysmall) [reply src/Lykke.Job.Stellar.Api/Startup.cs]: We use it synchronously anyway:
`appLifetime.ApplicationStarted.Register(() => StartApplication().GetAwaiter().GetResult());`
so, I considered it to be a bit less messy if the method was synchronous by itself using `GetAwaiter().GetResult()` inside:
`appLifetime.ApplicationStarted.Register(StartApplication);`

File: src/Lykke.Job.Stellar.Api/Startup.cs (hunk 3 above)
Comment (dmitrysmall) [reply src/Lykke.Job.Stellar.Api/Startup.cs]: The same as above.

File: src/Lykke.Job.Stellar.Api/Startup.cs (hunk 2 above)
Comment (KonstantinRyazantsev) [reply src/Lykke.Job.Stellar.Api/Startup.cs]: The rule is - if method contains async calls make it async too. And pull this async chain up throght call stack until it possible.

**Pull request merged**